### PR TITLE
30 Day Warranty

### PR DESCRIPTION
**Pull request by gruetter**
This is the 30 Day Warranty Pattern put in practice by the PayPal folks.

### CODE REVIEW

File: 30-day-warranty.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+A team developing a component which is used throughout an organization is
+resisting to accept or rejects contributions (feature requests) and as a result
+blocks progress or is disrupted by frequent escalations.
+
+- TBD: link to pattern "setting clear expectations for contributing code"
+
+# Context
+
+Teams depends on another team accepting their contributions so that a component
```
Comment (cewilliams) [inline 30-day-warranty.md]: I think this applies whether the contributing team provides the code or not. The end result that the contributing team is trying to achieve is that the feature is available to them to use, whether they, the receiving team, or another party write the code that provides the feature.

File: 30-day-warranty.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+receiving team does not have the resources, knowledge, permission, inclination 
+to write the contributed component.
+
+# Forces
+
+- There is distrust of contributions due to a past history of cheating: teams
+  submitted half finished contributions and subsequently filed requests for 
+  fixes that make it ready for use in production.
+- If code is contributed from outside the team, the team has the natural 
+  suspicition that the other team does not know how to write code that would
```
Comment (cewilliams) [inline 30-day-warranty.md]: suspicition -> suspicion

File: 30-day-warranty.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  meet the teams expectations.
+- Direction of loyalty: ?
+- There is a natural aversion to taking responsibility for code not written
+  by oneself.
+- Contributed needs to be heavily rewritten before being accepted into the
+  codebase.
+- There is the fear of the contributors not being available for support with
+  fixing bugs after the time on contribution.
+
```
Comment (cewilliams) [inline 30-day-warranty.md]: Additional forces may be:
- Teams fear contributed code will lead to high(er) maintenance costs but do not know how to control for that
- Receiving teams may fear that teaching others how to contribute code will expose technical debt in their system and that visibility may be damaging 
- Receiving teams may not believe that they will get acceptable code no matter how much mentoring they provide
- Either team may not feel confident in measuring risks or certifying that they are mitigated in a contribution

File: 30-day-warranty.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+a) clear contribution guidelines spelling out the expectations of the receiving
+   team and
+
+# Resulting Context
+
+- the receiving team is willing to accept contributions and able to share the
+  workload of initial adaptations/fixes
+- increased transparency and fairness
+- keeps escalations from becoming to heavyweight
```
Comment (cewilliams) [inline 30-day-warranty.md]: to -> too

File: 30-day-warranty.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+PayPal
+
+# Authors
+
+- Cedric Williams
+
+# Acknowledgement
+
+- Dirk
```
Comment (cewilliams) [inline 30-day-warranty.md]: Dirk ->  Dirk-Willem van Gulik

**Review by cewilliams — CHANGES_REQUESTED** [review]
Generally good, thanks for getting this in! Grammar changes are requested. Context and Forces changes are open for discussion.

File: 30-day-warranty.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+# Solution
+
+Address the fears of both the receiving and the contributing teams by 
+establishing a 30 day period starting the with time the contributed code
+goes into production, during which the contributing team consents to provide
+bug fixes to the receiving team.
+
+a) clear contribution guidelines spelling out the expectations of the receiving
+   team and
```
Comment (NewMexicoKid) [inline 30-day-warranty.md]: and ...?

**Comment by gruetter** [comment]
Thanks for the suggestions and changes, @cewilliams ! @psudars : will you have time to review, as well or would you like me to ask around if s.o. else would like to jump in your stead?

**Review by NewMexicoKid — APPROVED** [review]
This has been reviewed by the PLoP 2017 author team and assigned shepherd from PLoP.

**Pull request merged**